### ISSUE TITLE
illegal characters

### ISSUE THREAD

**Issue by techsideofficial** [ata4/bspsrc#177]
```
[error] Error occurred decompiling 'C:\Users\wiene\Downloads\rp_mk_city17_v2.bsp'
java.nio.file.InvalidPathException: Illegal char <> at index 34: materials/models/ff_props/oleaves  :>?8O.vtf
	at sun.nio.fs.WindowsPathParser.normalize(Unknown Source) ~[?:?]
	at sun.nio.fs.WindowsPathParser.parse(Unknown Source) ~[?:?]
	at sun.nio.fs.WindowsPathParser.parse(Unknown Source) ~[?:?]
	at sun.nio.fs.WindowsPath.parse(Unknown Source) ~[?:?]
	at sun.nio.fs.WindowsFileSystem.getPath(Unknown Source) ~[?:?]
	at java.nio.file.Path.resolve(Unknown Source) ~[?:?]
	at info.ata4.bspsrc.lib.PakFile.unpack(PakFile.java:83) ~[bspsrc.jar:?]
	at info.ata4.bspsrc.decompiler.BspSource.decompile(BspSource.java:172) ~[bspsrc.jar:?]
	at info.ata4.bspsrc.decompiler.BspSource.decompile(BspSource.java:128) ~[bspsrc.jar:?]
	at info.ata4.bspsrc.decompiler.BspSource.lambda$run$0(BspSource.java:94) ~[bspsrc.jar:?]
	at java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(Unknown Source) [?:?]
	at java.util.concurrent.ForkJoinTask.doExec(Unknown Source) [?:?]
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source) [?:?]
	at java.util.concurrent.ForkJoinPool.scan(Unknown Source) [?:?]
	at java.util.concurrent.ForkJoinPool.runWorker(Unknown Source) [?:?]
	at java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source) [?:?]
```

**Comment by worMatty** [comment]
You will put people off helping you if you are rude to them.

**Comment by techsideofficial** [comment]
> You will put people off helping you if you are rude to them.

fixed

**Comment by techsideofficial** [comment]
I changed the name

**Comment by worMatty** [comment]
Since this appears to be an issue with unpacking custom assets, as a workaround you might want to try only decompiling the VMF and not extracting the assets. Then if you need the packed content, try extracting it using GCFScape and/or 7-Zip/Nanazip.

Both apps can be installed using `winget` in PowerShell if you're on Windows 11.

**Comment by rihi** [comment]
This is caused by some embedded file containing invalid characters, causing the writing of the file to fail. I fixed this with 63027644 by just skipping these files for now.
As a workaround, until the fix is released with next version, you can just do what @worMatty suggested.
